drivers/device_uid/stm32_uid.c

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2018 Alexander Wachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <soc.h>
+#include <device_uid.h>
+#include <string.h>
+
+#if !defined(CONFIG_SOC_SERIES_STM32F0X) && \
+	!defined(CONFIG_SOC_SERIES_STM32F3X) && \
+	!defined(CONFIG_SOC_SERIES_STM32L4X) && \
+	!defined(CONFIG_SOC_SERIES_STM32F7X)
+#error UNIQUE ID only available on STM32F0, STM32F3, STM32l4 and STM32F7 series
+#endif
+
+struct stm32_uid {
+	u32_t id[3];
+};
+
+static int stm32_uid_init(struct device *dev)
+{
+	return 0;
+}
+
+static size_t stm32_uid_get(struct device *dev, u8_t *buffer, size_t length)
+{
+	struct stm32_uid dev_id;
+
+	if (length > sizeof(dev_id.id)) {
+		length = sizeof(dev_id.id);
+	}
+
+	dev_id.id[0] = HAL_GetUIDw0();
+	dev_id.id[1] = HAL_GetUIDw1();
+	dev_id.id[2] = HAL_GetUIDw2();
+
+	memcpy(buffer, dev_id.id, length);
```

Comment:
If the range check on `length` were immediately before the use of `length` people wouldn't be tempted to comment "Wait, you didn't check  for source overrun."